__tests__/app.test.js

```diff
@@ -241,37 +241,165 @@ describe("APP", () => {
+                expect(article).toEqual({
+                  article_id: expect.any(Number),
+                  title: expect.any(String),
+                  topic: "cats",
+                  author: expect.any(String),
+                  body: expect.any(String),
+                  created_at: expect.any(String),
+                  comment_count: expect.any(String),
+                  article_img_url: expect.any(String),
+                  votes: expect.any(Number),
+                });
+              });
+            });
+        });
+        test("Query to sort by any column", () => {
+          return request(app)
+            .get("/api/articles/?sort=comment_count")
+            .expect(200)
+            .then(({ body }) => {
+              for (let i = 0; i < body.articles.length - 1; i++)
+                expect(+body.articles[i].comment_count).toBeLessThanOrEqual(
+                  +body.articles[i + 1].comment_count
+                );
+            });
+        });
+        test("Query to sort by any column 2", () => {
+          return request(app)
+            .get("/api/articles/?sort=votes")
+            .expect(200)
+            .then(({ body }) => {
+              for (let i = 0; i < body.articles.length - 1; i++)
+                expect(+body.articles[i].votes).toBeLessThanOrEqual(
+                  +body.articles[i + 1].votes
+                );
+            });
+        });
+        test("Query to order by decending", () => {
+          return request(app)
+            .get("/api/articles/?order=asc")
+            .expect(200)
+            .then((({ body }) => {
+              for (let i = 0; i < body.articles.length - 1; i++)
+                expect(new Date(body.articles[i].created_at)).toBeBefore(
+                  new Date(body.articles[i + 1].created_at)
+                );
+            })
+            );
+        });
+        test("If query is ommited all articles are returned", () => {
+          return request(app)
+            .get("/api/articles/?topic=")
+            .expect(200)
+            .then((({ body }) => {
+              expect(body.articles).toHaveLength(12)
+              body.articles.forEach((article) => {
+                expect(article).toEqual({
+                  article_id: expect.any(Number),
+                  title: expect.any(String),
+                  topic: expect.any(String),
+                  author: expect.any(String),
+                  body: expect.any(String),
+                  created_at: expect.any(String),
+                  comment_count: expect.any(String),
+                  article_img_url: expect.any(String),
+                  votes: expect.any(Number),
+                });
+              });
+            })
+            );
+        });
+        test("error handling - should return 400 error with bad request / SQL injection attempt", () => {
+          return request(app)
+            .get("/api/articles/?topic=qfq33ofoqbfq")
```

Comment:
I'm undecided about this test; thing jibberish string could theoretically be a legitimate topic, so maybe 400 isn't appropriate here. 404 topic not found, or an empty array, perhaps?
It's your choice